app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :validatable
+
+  enum gender: %i[masculino femenino]
+
+  validates :username, :gender, :password_confirmation, presence: true
+  validates :username, length: { maximum: 30 }
+  validates :gender, inclusion: { in: genders.keys }, if: :gender_present?
```

Comment:
You can change the `if: :gender_present?` to `if: Proc.new { |u| u.gender.present? }`. That way you won't need the private method